development/build/static.js

```diff
@@ -0,0 +1,87 @@
+const watch = require('gulp-watch')
+// const endOfStream = require('pify')(require('end-of-stream'))
+const pify = require('pify')
+const { exec } = pify(require('child_process'))
+
+const { createTask, taskParallel } = require('./task')
+
+module.exports = createStaticAssetTasks
+
+
+const copyTargets = [
+  {
+    src: `./app/_locales/`,
+    dest: `_locales`,
+  },
+  {
+    src: `./app/images/`,
+    dest: `images`,
+  },
+  {
+    src: `./node_modules/eth-contract-metadata/images/`,
+    dest: `images/contract`,
+  },
+  {
+    src: `./app/fonts/`,
+    dest: `fonts`,
+  },
+  {
+    src: `./app/vendor/`,
+    dest: `vendor`,
+  },
+  {
+    src: `./ui/app/css/output/`,
+    pattern: `*.css`,
+    dest: ``,
+  },
+  {
+    src: `./app/`,
+    pattern: `*.html`,
+    dest: ``,
+  },
+]
+
+const copyTargetsDev = [
+  ...copyTargets,
+  {
+    src: './app/scripts/',
+    pattern: '/chromereload.js',
+    dest: ``,
+  },
+]
+
+function createStaticAssetTasks ({ livereload, browserPlatforms }) {
+
+  const prod = createTask('static:prod', taskParallel(...copyTargets.map((target) => {
+    return function copyStaticAssets () {
+      return performCopy(target)
+    }
+  })))
+  const dev = createTask('static:dev', taskParallel(...copyTargetsDev.map((target) => {
+    return function copyStaticAssets () {
+      return setupLiveCopy(target)
+    }
+  })))
+
+  return { dev, prod }
+
+  function setupLiveCopy (target) {
+    const pattern = target.pattern || '/**/*'
+    watch(target.src + pattern, (event) => {
+      livereload.changed(event.path)
+      performCopy(target)
+    })
+    performCopy(target)
+  }
+
+  async function performCopy (target) {
+    await Promise.all(browserPlatforms.map(async (platform) => {
+      if (target.pattern) {
+        await exec(`cp ${target.src}${target.pattern} ./dist/${platform}/${target.dest}`)
```

Comment:
I don't think this would work on Windows, whereas our current build does.

Perhaps we could go back to using `gulp.src` for now, or a combination of `globby` and `copy` from `fs-extra`? 🤔

Reply:
gulp.src was 150x slower for copying... didnt really affect the actual time to completion but it felt silly
fs-extra may have what we need

also, windows, meh?

Reply:
I'd prefer that our dev environment work correctly on WIndows. A _lot_ of developers use Windows, which could include current or future team members, but _definitely_ includes contributors. If we want to make this project friendly for community contributions, then Windows support is a must.

Plus it's pretty low-cost - a few seconds more work in copying or another small dev dependency, not a huge deal.

Reply:
might be nice to run our build on windows in CI to ensure that this works and is not regressed. but i dont think windows is supported (ha!)

Reply:
went with `fs.promises.copyFile` + `fast-glob`